django/forms/models.py

```diff
@@ -1025,7 +1031,10 @@ def _get_foreign_key(parent_model, model, fk_name=None, can_fail=False):
             f for f in opts.fields
             if isinstance(f, ForeignKey) and (
                 f.remote_field.model == parent_model or
-                f.remote_field.model in parent_model._meta.get_parent_list()
+                f.remote_field.model in parent_model._meta.get_parent_list() or (
+                    f.remote_field.model and
+                    f.remote_field.model._meta.proxy_for_model in parent_model._meta.get_parent_list()
+                )
```

Comment:
We're checking `get_parent_list()` in a loop so I would cache it in the variable:
```python
        parent_list = parent_model._meta.get_parent_list()
        fks_to_parent = [
            f for f in opts.fields
            if isinstance(f, ForeignKey) and (
                f.remote_field.model == parent_model or
                f.remote_field.model in parent_list or (                       
                    f.remote_field.model._meta.proxy and
                    f.remote_field.model._meta.proxy_for_model in parent_list
                )   
            )   
        ]
```